src/dune_rules/mdx.ml

```diff
@@ -147,8 +150,13 @@ let decode =
        field "files" Predicate_lang.Glob.decode ~default:default_files
      and+ packages =
        field ~default:[] "packages" (repeat (located Package.Name.decode))
-     and+ preludes = field ~default:[] "preludes" (repeat Prelude.decode) in
-     { loc; files; packages; preludes })
+     and+ preludes = field ~default:[] "preludes" (repeat Prelude.decode)
+     and+ enabled_if =
+       let open Enabled_if in
+       let allowed_vars = common_vars ~since:(2, 9) in
```

Comment:
I wasn't sure what to pass here. Would `Any` make sense? it seems to be a compatibility problem since users could put recent variables that don't exist on older ones. Or is there a special versioning mechanism for this?

Reply:
You can do the same as for the `rule` stanza. We pass `Any` there, so `Any` here seems fine. We only use use a restricted set in places where we need to evaluate the `enabled_if` early on, such as for `install` or `executables`. In these cases, we need to evaluate the `enabled_if` early on in order to know what will be present in the bin directory, which is needed for resolving program names.

Reply:
Thanks, I did that